core/integer/pow_spec.rb

```diff
@@ -0,0 +1,136 @@
+      2.pow(63).should eql 9223372036854775808
+      2.pow(64).should eql 18446744073709551616
+      8.pow(23).should eql 590295810358705651712
+    end
+
+    it "raises negative numbers to the given power" do
+      (-2).pow(29).should eql(-536870912)
+      (-2).pow(30).should eql(1073741824)
+      (-2).pow(31).should eql(-2147483648)
+      (-2).pow(32).should eql(4294967296)
+
+      (-2).pow(61).should eql(-2305843009213693952)
+      (-2).pow(62).should eql(4611686018427387904)
+      (-2).pow(63).should eql(-9223372036854775808)
+      (-2).pow(64).should eql(18446744073709551616)
+    end
+
+    it "can raise 1 to a bignum safely" do
+      1.pow(4611686018427387904).should eql 1
+    end
+
+    it "can raise -1 to a bignum safely" do
+      (-1).pow(4611686018427387904).should eql(1)
+      (-1).pow(4611686018427387905).should eql(-1)
+    end
+
+    it "returns Float::INFINITY when the number is too big" do
+      2.pow(27387904).class.should == Integer
+      2.pow(427387904).class.should == Float
+      2.pow(427387904).should == Float::INFINITY
+    end
+
+    it "raises a ZeroDivisionError for 0 ** -1" do
+      -> { 0.pow(-1) }.should raise_error(ZeroDivisionError)
+      -> { 0.pow(Rational(-1, 1)) }.should raise_error(ZeroDivisionError)
+    end
+
+    it "returns Float::INFINITY for 0 ** -1.0" do
+      0.pow(-1.0).should == Float::INFINITY
+    end
+
+    it "raises a TypeError when given a non-numeric power" do
+      -> { 13.pow("10") }.should raise_error(TypeError)
+      -> { 13.pow(:symbol) }.should raise_error(TypeError)
+      -> { 13.pow(nil) }.should raise_error(TypeError)
+    end
+
+    it "coerces power and calls #**" do
+      num_2 = mock("2")
+      num_13 = mock("13")
+      num_2.should_receive(:coerce).with(13).and_return([num_13, num_2])
+      num_13.should_receive(:**).with(num_2).and_return(169)
+
+      13.pow(num_2).should == 169
+    end
+
+    it "returns Float when power is Float" do
+      2.pow(2.0).should == 4.0
+    end
+
+    it "returns Rational when power is Rational" do
+      2.pow(Rational(2, 1)).should == Rational(4, 1)
+    end
+
+    it "returns a complex number when negative and raised to a fractional power" do
+      (-8).pow(1.0/3)         .should be_close(Complex(1, 1.73205), TOLERANCE)
+      (-8).pow(Rational(1, 3)).should be_close(Complex(1, 1.73205), TOLERANCE)
+    end
+
+    context "second argument is passed" do
+      it "returns modulo of self raised to the given power" do
+        2.pow(5, 12).should == 8
+        2.pow(6, 13).should == 12
+        2.pow(7, 14).should == 2
+        2.pow(8, 15).should == 1
+      end
+
+      it "doesn't work well with bignum" do
+        2.pow(61, 213693951).should eql(147138137)
+        2.pow(61, 9213693951).should eql 1
+
+        2.pow(61, 5843009213693951).should eql 1
+        2.pow(62, 5843009213693952).should eql 1
+        2.pow(63, 5843009213693953).should eql 1
+        2.pow(64, 5843009213693954).should eql 1
+      end
```

Comment:
looks like a bug:
`2.pow(61, 9213693951) # => 1`
but
`(2 ** 61) % 9213693951 # => 5432161352`

Reply:
Yes indeed, could you add the example to https://bugs.ruby-lang.org/issues/14259?

Reply:
In this PR, you can then specify the right behavior (`eql 147138137`) and guard it with:
```ruby
ruby_bug '#13669', '2.5'...'2.5.1' do
  ...
```